### PR TITLE
Correct documentation examples of `optimizer_step`

### PR DESCRIPTION
**Pull request by willprice**
## What does this PR do?
Without the default arguments set in `optimizer_step` the examples fail due to the arguments not being provided

Fixes # (issue)

# Before submitting

- [-] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [-] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

### CODE REVIEW

**Review by ananyahjha93 — APPROVED** [review]

File: docs/source/optimizers.rst [hunk 1]
```diff
@@ -85,14 +85,14 @@ For example, here step optimizer A every 2 batches and optimizer B every 4 batch
 
 .. testcode::
 
-    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_i, second_order_closure, on_tpu, using_native_amp, using_lbfgs):
+    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_i, second_order_closure=None, on_tpu=False, using_native_amp=False, using_lbfgs=False):
```
Comment (rohitgr7) [inline docs/source/optimizers.rst:88]:
```suggestion
    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, second_order_closure=None, on_tpu=False, using_native_amp=False, using_lbfgs=False):
```

File: docs/source/optimizers.rst [hunk 2]
```diff
@@ -85,14 +85,14 @@ For example, here step optimizer A every 2 batches and optimizer B every 4 batch
...
         optimizer.step()
 
     def optimizer_zero_grad(self, current_epoch, batch_idx, optimizer, opt_idx):
       optimizer.zero_grad()
 
     # Alternating schedule for optimizer steps (ie: GANs)
-    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_i, second_order_closure, on_tpu, using_native_amp, using_lbfgs):
+    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_i, second_order_closure=None, on_tpu=False, using_native_amp=False, using_lbfgs=False):
```
Comment (rohitgr7) [inline docs/source/optimizers.rst:95]:
```suggestion
    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, second_order_closure=None, on_tpu=False, using_native_amp=False, using_lbfgs=False):
```

File: docs/source/optimizers.rst [hunk 3]
```diff
@@ -113,7 +113,7 @@ Here we add a learning-rate warm up
 .. testcode::
 
     # learning rate warm-up
-    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_i, second_order_closure, on_tpu, using_native_amp, using_lbfgs):
+    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_i, second_order_closure=None, on_tpu=False, using_native_amp=False, using_lbfgs=False):
```
Comment (rohitgr7) [inline docs/source/optimizers.rst:116]:
```suggestion
    def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, second_order_closure=None, on_tpu=False, using_native_amp=False, using_lbfgs=False):
```

**Review by rohitgr7 — APPROVED** [review]
looks good! Just few minor updates :)

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**